### DISCUSSION TITLE
Megamenu – which component to use?

### DISCUSSION THREAD

**Discussion by lapidus** [chakra-ui/chakra-ui#4567] [Q&A] [answered]
Some of my Desktop menu items need a large PopOver kind of menu like the attached examples.
The menu always takes up a certain width and the trigger is more or less disconnected from the actual Drawer. It is important that the menu works as the PopOver in the sense that it closes onBlur.

Which component is it best to base this design on?
- Popover with static width and no arrow?
- Drawer with Hover trigger?
- Custom box with useDisclosure?
- Other!? 😅 

Thanks! 

![image](https://user-images.githubusercontent.com/66601/129392648-988bd683-14bc-48ea-9c3a-cd513a011beb.png)
![image](https://user-images.githubusercontent.com/66601/129392674-87522d4e-900f-403d-936b-f10273f1c772.png)
![image](https://user-images.githubusercontent.com/66601/129392779-18df1602-53a6-4488-b10a-299b1badfcd4.png)

**Comment by abhishekmg** [comment]
I think the `Hasura` example can be achieved just by using css (hover). Can be achieved by combining `Box` component from charka. The `Box` needs to render on hover of the nav option. 

This is one way the menu can be achieved .

**Reply by lapidus** [reply]
Thanks, I am still a bit concerned about accessibility and onBlur of such an approach? It seems PopOver handles these things naturally but the placement/trigger is not really useful for this scenario? Would be great with an example if anyone has!

**Reply by abhishekmg** [reply]
@lapidus sorry I should have given example for the css option earlier but here one simple implantation 
https://codesandbox.io/s/jovial-brahmagupta-fu1he?file=/src/index.js

**Comment by anubra266** [comment, marked as answer]
Check the implementation on  [Choc UI's With Full-Width Flyouts Header](https://choc-ui.tech/docs/elements/headers#headers/wfwf)

**Reply by lapidus** [reply]
Thank you both @anubra266, @abhishekmg – great examples! 🙏

**Comment by lapidus** [comment]
For anyone coming after me here, "Group Hover" (https://vue.chakra-ui.com/pseudobox#group-hover) as pointed above, is your friend 👍
